README.md

```diff
@@ -1,3 +1,76 @@
 # json-schema-to-nickel
 
-Convert JSON schemas into Nickel contracts. WIP.
+A tool to convert [JSON schema](https://json-schema.org) into [Nickel](https://www.nickel-lang.org) contracts.
+
+## How to use it?
+
+`json-schema-to-nickel` is a standalone binary that reads a JSON schema file and
+outputs a Nickel contract. You can run it from source using `cargo` by running
+
+```shell
+cargo run -- <path to your schema.json>
+```
+
+in a checkout of this repository. Alternatively, you can use `nix`:
+
+```shell
+nix run github:nickel-lang/json-schema-to-nickel -- <path to your schema.json>
+```
+
+anywhere. This will retrieve the latest revision from GitHub, build
+`json-schema-to-nickel` and run the resulting binary on your JSON schema.
+
+Either invocation will print the generated Nickel contract to standard
+output. It can get rather large, so it's recommended to redirect the output into
+a file, for example by using
+
+```shell
+cargo run -- schema.json > contract.ncl
+```
+
+Once you have a output of `json-schema-to-nickel` in a file, say `contract.ncl`,
+you can check a Nickel configuration against it by importing the file by its
+path, e.g.
+
+```nickel
+let
+  SchemaContract = import "./contract.ncl"
+in {
+  foo = "bar"
+} | SchemaContract
+```
+
+## How it works
+
+We're using the [schemars][schemars] crate to parse
+JSON schema. A JSON schema encodes a predicate on JSON objects, that is, a
+function from JSON objects returning `true` or `false`. Such a function can also
+be written in Nickel and a big part of `json-schema-to-nickel` is mechanically
+generating such a predicate using a [Nickel library](./lib/predicates.ncl).
+However, such predicates are not ideal for use in Nickel and many aspects
+of JSON schema can be encoded as Nickel record contracts. The second major
+part of `json-schema-to-nickel` is figuring out when a JSON schema can be
+represented as a record contract, and generating this contract when it makes
+sense. Finally, there is some housekeeping to do for dispatching between these
+two functionalities.
```

Comment:
1. I would make it clear that this is not currently optimal. There are certainly json schemas that could be turned into record contracts that we don't (yet).
2. I would also say why we want to generate contracts this way. i.e. laziness, and LSP integration. I see you explain it below, but I think it belongs in this section too.